webhook/resources/volume/validator.go

```diff
@@ -126,19 +126,17 @@ func (v *volumeValidator) Create(request *admission.Request, newObj runtime.Obje
 		return werror.NewInvalidError(err.Error(), "")
 	}
 
+	dataEngineSetting := types.SettingNameV1DataEngine
 	if datastore.IsBackendStoreDriverV2(volume.Spec.BackendStoreDriver) {
-		v2DataEngineEnabled, err := v.ds.GetSettingAsBool(types.SettingNameV2DataEngine)
-		if err != nil {
-			err = errors.Wrapf(err, "failed to get spdk setting")
-			return werror.NewInvalidError(err.Error(), "")
-		}
-		if !v2DataEngineEnabled {
-			return werror.NewInvalidError("v2 data engine is not enabled", "")
-		}
-
-		if volume.Spec.Frontend == longhorn.VolumeFrontendISCSI {
-			return werror.NewInvalidError("v2 data engine does not support iSCSI frontend", "")
-		}
+		dataEngineSetting = types.SettingNameV2DataEngine
+	}
+	enabled, err := v.ds.GetSettingAsBool(dataEngineSetting)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to get %v setting", dataEngineSetting)
+		return werror.NewInvalidError(err.Error(), "")
+	}
+	if !enabled {
+		return werror.NewInvalidError(fmt.Sprintf("%v data engine is not enabled", dataEngineSetting), "")
 	}
```

Comment:
nit: The validation in volume/engine/replica is almost the same. Suggest extracting that part into `webhook/resources` package instead.

Reply:
Addressed. Thank you.